lib/go/thrift/server_socket.go

```diff
@@ -68,15 +68,18 @@ func (p *TServerSocket) Listen() error {
 
 func (p *TServerSocket) Accept() (TTransport, error) {
 	p.mu.RLock()
-	defer p.mu.RUnlock()
+	listener := p.listener
+	interrupted := p.interrupted
+	p.mu.RUnlock()
 
```

Comment:
Agree. Looks strange. and the comment even says

      // Protects the interrupted value to make it thread safe.

What about moving the ```listener := p.listener``` line down just before it is used? Or do I overlook sth?